### PR TITLE
Move code into the cleanup subroutine to reduce duplication

### PR DESCRIPTION
**Pull request by b10n1k**
Refactor recent code related to publiccloud clean subroutine.                                                                                                                                                                                
As for now we get a tunnel-console to the openqa instance                                                                                                                                                                                    
and exit the ssh terminal using manul calls before run the actual cleanup.                                                                                                                                                                   
The idea is that the cleanup can be reused and we dont need to                                                                                                                                                                               
repeat code.                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                             
i found easier and cleaner to reuse the `select_host_console` than any other approach.

- Related ticket: https://progress.opensuse.org/issues/80922
- [Verification run sle15](https://openqa.suse.de/tests/overview?distri=sle&version=15-SP2&build=b10n1k%2Fos-autoinst-distri-opensuse%2311711)
-  [Verification run sle12](https://openqa.suse.de/tests/overview?version=12-SP5&distri=sle&build=b10n1k%2Fos-autoinst-distri-opensuse%2311711)

### CODE REVIEW

File: lib/publiccloud/ec2.pm [hunk 1]
```diff
@@ -169,6 +169,9 @@ sub img_proof {
 
 sub cleanup {
     my ($self) = @_;
+    select_console 'tunnel-console', await_console => 0;
```
Comment (cfconrad) [inline lib/publiccloud/ec2.pm:172]: AFAIK, I wouldn't select a console in this class. The console depends on the test-module and here we use this tunnel and serial_terminal.
So from my side this break the logic!

**Review by cfconrad — CHANGES_REQUESTED** [review]

File: lib/publiccloud/ec2.pm (hunk 1 above)
Comment (b10n1k) [reply lib/publiccloud/ec2.pm]: @cfconrad i made some changes, removing this particular addings from the ec2.pm. i tried to reuse the `select_host_console`. i will resolve this and feel free to comment to the new commits

File: lib/publiccloud/ssh_interactive_init.pm [hunk 2]
```diff
@@ -19,15 +19,13 @@
...
+use publiccloud::utils;
 use strict;
 use warnings;
 use testapi;
 
 sub post_fail_hook {
-    select_console 'tunnel-console', await_console => 0;
-    send_key "ctrl-c";
-    send_key "ret";
+    select_host_console();
```
Comment (cfconrad) [inline lib/publiccloud/ssh_interactive_init.pm:28]: You are skipping `CTRL+C` here. Not sure if it is a problem. But I guess the reason is, if there is something running on that console, it should be terminated, before typing the next commands.

Comment (cfconrad) [reply lib/publiccloud/ssh_interactive_init.pm:28]: @pdostal wdyt?

Comment (cfconrad) [reply lib/publiccloud/ssh_interactive_init.pm:28]: Ah ok, overlooked this https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11711/files#diff-27d3901d8dece7c398b9abdb53c98d17bad7e5c1cf9ed74f14ed9e68a6e6a1c2R34

For me it looks ok. I think an additional `Ctrl+c` and `enter` should not harm.

**Review by cfconrad — APPROVED** [review]
LGTM, 
@pdostal plz take a look too

File: lib/publiccloud/ssh_interactive_init.pm (hunk 2 above)
Comment (b10n1k) [reply lib/publiccloud/ssh_interactive_init.pm:28]: yea. this is what i was thinking too. i was not sure if i had to comment tho to not get people confused of why this is there and get it deleted by accident

**Review by grisu48 — APPROVED** [review]
LGTM! Perhaps we can do a VR for one of the SLE12 variants? E.g. SLE12-SP5 EC2-BOYS?

**Comment by pdostal** [comment]
> LGTM,
> @pdostal plz take a look too

Thank you Clemens. Didn't have enough time to do so 😞

**Comment by dzedro** [comment]
Looks like `await_console => 0` had purpose.
https://openqa.suse.de/tests/5272317#step/ssh_interactive_end/3

**Comment by b10n1k** [comment]
> Looks like `await_console => 0` had a purpose.
> https://openqa.suse.de/tests/5272317#step/ssh_interactive_end/3

I am sorry @dzedro i have reverted this for the time being https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11746

**Pull request merged**